source/slang/slang-state-serialize.cpp

```diff
@@ -1072,10 +1074,234 @@ struct LoadContext
+                {
+                    const auto& srcOutputState = base.asRaw(srcOutputStateOffset);
+
+                    SLANG_ASSERT(srcOutputState.entryPointIndex < requestState->entryPoints.getCount());
+
+                    String entryPointPath;
+                    if (srcOutputState.outputPath)
+                    {
+                        entryPointPath = base.asRaw(srcOutputState.outputPath)->getSlice();
+                    }
+
+                    dstTargetInfo->entryPointOutputPaths.Add(srcOutputState.entryPointIndex, entryPointPath);
+                }
+            }
+#endif
+        }
+    }
+
+    {
+        const auto& srcPaths = requestState->searchPaths;
+        for (Index i = 0; i < srcPaths.getCount(); ++i)
+        {
+            cmd.addArg("-I");
+            cmd.addArg(base.asRaw(base.asRaw(srcPaths[i]))->getSlice());
+        }
+    }
+
+    {
+        for (const auto& define : requestState->preprocessorDefinitions)
+        {
+            StringBuilder builder;
+            builder << "-D" << base.asRaw(base.asRaw(define).first)->getSlice();
+            if (base.asRaw(define).second)
+            {
+                builder << "=" << base.asRaw(base.asRaw(define).second)->getSlice();
+            }
+        }
+    }
+
+    {
+        const auto& srcTranslationUnits = requestState->translationUnits;
+
+        for (Index i = 0; i < srcTranslationUnits.getCount(); ++i)
+        {
+            const auto& srcTranslationUnit = base.asRaw(srcTranslationUnits[i]);
+
+            //context.loadDefines(srcTranslationUnit.preprocessorDefinitions, dstTranslationUnit->preprocessorDefinitions);
+
+#if 0
+            if (srcTranslationUnit.moduleName)
+            {
+                moduleName = base[srcTranslationUnit].moduleName->getSlice());
+            }
+#endif
+      
+            const auto& srcSourceFiles = srcTranslationUnit.sourceFiles;
+            
+            for (Index j = 0; j < srcSourceFiles.getCount(); ++j)
+            {
+                SourceFileState* sourceFile = base.asRaw(base.asRaw(srcSourceFiles[i]));
+                OffsetString* path = base[sourceFile->foundPath];
+
+                if (path)
+                {
+                    cmd.addArg(path->getSlice());
+                }
+            }
+        }
+    }
+
+    // Entry points
+    {
+        for (const auto& srcEntryPointOffset : requestState->entryPoints)
+        {
+            const auto srcEntryPoint = base.asRaw(srcEntryPointOffset);
+
+            const char* name = srcEntryPoint.name ? base.asRaw(srcEntryPoint.name)->getCstr() : nullptr;
+
+            cmd.addArg("-entry");
+            cmd.addArg(name);
+
+            cmd.addArg("-stage");
+            UnownedStringSlice stageText = getStageText(srcEntryPoint.profile.GetStage());
+            cmd.addArg(stageText);
+
+            //cmd.addArg("-profile");
+            //cmd.addArg(Profile(srcEntryPoint.profile).getName());
```

Comment:
The fact that we store a full `Profile` on an entry point is at this point a legacy/compatibility feature, and not something we need to account for here. The `-profile` option applies to a target now, and that is intentional.

There is some logic in the options parsing to try and tease apart something like `-profile ps_5_0` to treat it as if the user specified `-state fragment -profile sm_5_0`, but that isn't something that we'd want the repro functionality to ever exploit.